### DISCUSSION TITLE
What is the TypeScript type of App Router (useRouter)?

### DISCUSSION THREAD

**Discussion by don-esteban** [vercel/next.js#60064] [Help] [answered]
### Summary

I pass down the App Router to child components and would like (should I?) to declare the specific TypeScript types.

1. Is this best practice?
2. Where do I find the type?

`function Products({cart, router}: {cart: Cart, router: any}) {...} `

### Additional information

```js
import { useRouter } from "next/navigation";

export default function ConfirmOrderPage() {
    const router = useRouter();
    ...
    return (
        <Products cart={cart} router={router} />
    )   
}

function Products({cart, router}: {cart: Cart, router: any}) {...}
```

**Comment by mzuleta6** [comment, marked as answer]
The right type for router is "AppRouterInstance" @don-esteban and you don't should pass router as Props this isn't a good practice, call the hook inside the component.

**Reply by don-esteban** [reply]
Thank you. I was also considering 'best practices' as you mentioned. Upon reflection, you are right! And I no longer need the type. However, out of curiosity, where is the AppRouterInstance located (for `import`)?

**Reply by bcholokava** [reply]
@mzuleta6 Hi, Regarding your comment about not passing the router as props, I've been exploring an approach where we locate some helper functions outside of a component, which use the router, and we pass the router as a parameter to these functions. Could you share your thoughts on this practice?

**Reply by bcholokava** [reply]
BTW, passing the router as a prop isn't good practice indeed. You can directly call the useRouter hook inside the component.

**Reply by mzuleta6** [reply]
@bcholokava I think in this case you don't need use a helper for router cause it's a tools included on Next. When you need some helper function you could be consider use a `utils` file like a place to put your functions used around your project. There are many practices to manage it but this is one the most common.

**Reply by adnjoo** [reply]
@mzuleta6 you can e.g. import the type from 
```ts
import { AppRouterInstance } from 'next/dist/shared/lib/app-router-context.shared-runtime';
```
